src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -39,15 +39,15 @@
     /** Default tab width for column reporting. */
     private static final int DEFAULT_TAB_WIDTH = 8;
 
+    /**
+     * The check context.
+     * @noinspection ThreadLocalNotStaticFinal
```

Comment:
> Suppose we have two instances (named foo and bar) of some FileSet class. We call foo.log(...) and then foo::messageCollector contains one message. However, after calling bar.log(...) the foo::messageCollector will contain two messages, because the field is static and we're inside the same thread. Not sure if this is the expected behavior.

Again, not sure if we can use static here. At least, there were some failed tests, when I changed this field to static